homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,149 @@
+"""Support for the WatchYourLAN service."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="online_status",
+        name="Online Status",
+        icon="mdi:lan-connect",
+    ),
+    SensorEntityDescription(
+        key="ip_address",
+        name="IP Address",
+        icon="mdi:ip-network",
+    ),
+    SensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        icon="mdi:lan",
+    ),
+    SensorEntityDescription(
+        key="iface",
+        name="Network Interface",
+        icon="mdi:ethernet",
+    ),
+]
```

Comment:
Please use translations and icon translations

Reply:
I got the icon tanslation working, but not the name translation. Do you have any pointers for where I'm going wrong here?